### PR TITLE
Pay for user CPU

### PR DESCRIPTION
**Pull request by n13**
## Spec

- Pay for user CPU if user is member of DAO
- Don't pay if user not member of DAO

## Concept
We parse esr codes as usual
The parser also checks if free CPU is possible and stores the action - for this it needs the account and the network
UX checks whether or not there's a free action and shows that
When we send the transaction off, we add the free action if we can, and not if we can't. 

This way we don't change anything about the parser, the EOS transaction, or the SeedsESR object, which is a very complex piece of code coming from an old ESR implementation which was ported to Flutter from JavaScript. 

## To do list
- [x] buildFreeTransaction service
- [x] FreeTransactionUseCase use case
- [x] Pay CPU call works as intended (tested on Telos)
- [x] The EosTransaction needs to be modified before it is showing on the signing screen - currently doing it in the wrong place, after signing. This means - it's signing, but not showing.
- [x] UX for free transaction as per Luigi

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# rename to .evn and fill in the secret values
```
Comment (n13) [inline .env.example:1]: add example file so it's less mysterious

File: lib/ui/sign_transaction/interactor/sign_transaction_bloc.dart [hunk 2]
```diff
@@ -33,9 +37,20 @@ class SignTransactionBloc extends Bloc<SignTransactionEvent, SignTransactionStat
...
   FutureOr<void> _onUserSlideCompleted(_OnUserSlideCompleted event, Emitter<SignTransactionState> emit) async {
+    /// Try to create a free transaction
+
+    /// Would be a lot nicer if we can get the authenticated user somewhere from the bloc
+    /// Would also work better for supporting multiple accounts.
+    ///
+    /// This is also the wrong place to try to create a free transaction, it needs to happen upstream when we show the data on
+    /// the transaction screen - this code here only gets executed when user signs and slides right
+    /// But the transaction transformation is now tested, and the execution is also tested.
+    final authenticated = GetIt.I.get<AuthRepository>().authDataOrCrash;
```
Comment (gguijarro-c-chwy) [inline lib/ui/sign_transaction/interactor/sign_transaction_bloc.dart:48]: inject the auth repo into the bloc.

**Comment by n13** [comment]
Ok this needs to be made so the sign transaction view already gets the changed / free transaction sequence, will do that

File: lib/ui/sign_transaction/interactor/sign_transaction_bloc.dart (hunk 2 above)
Comment (n13) [reply lib/ui/sign_transaction/interactor/sign_transaction_bloc.dart:48]: yeah well the entire code shouldn't be there... fixing this

**Comment by n13** [comment]
Changed - see initial description

File: lib/ui/home_page/usecases/parse_qr_code_use_case.dart [hunk 3]
```diff
@@ -4,45 +4,52 @@ import 'package:hypha_wallet/core/crypto/seeds_esr/scan_qr_code_result_data.dart
...
     if (qrCodeValidationResult.isValue) {
-      return HResult.Result.value(qrCodeValidationResult.asValue!.value);
+      final data = qrCodeValidationResult.asValue!.value;
+      data.freeCpuAction = await _freeTransactionUseCase.run(user, data.transaction.network);
+      return HResult.Result.value(data);
     } else {
       return HResult.Result.error(qrCodeValidationResult.asError!.error as HyphaError);
     }
   }
-}
```
Comment (n13) [inline lib/ui/home_page/usecases/parse_qr_code_use_case.dart:33]: the code below has not actually changed, just moved it inside the class - it was outside the class before, for whatever reason

**Review by gguijarro-c-chwy — APPROVED** [review]

**Comment by n13** [comment]
Ok I added a feature flag and tested it, shipping it now

The reason is, we don't need it on Telos and in case something goes wrong with this we can turn it off

**Pull request merged**